docs/other/group_by.md

```diff
@@ -0,0 +1,95 @@
+# How to avoid the `UserWarning` error while using `group_by`
+
+## Introduction
+
+If you have ever experienced the `UserWarning: Found complex group-by expression, which can't be expressed efficiently with the pandas API. If you can, please rewrite your query such that group-by aggregations are simple (e.g. mean, std, min, max, ...)` message while using the narwhals `group_by()` method this is for you. If you haven't, this is also for you as you might experience it and you need to know how to avoid it.
+
+The `UserWarning` is thrown for exactly the reason that is given. The pandas API most likely cannot efficiently handle the complexity of the `group_by()` operations you are trying to run. Take the following two codes as an example.
+
+When you run:
+=== "Approach 1"
+    ```python exec="0" source="above" result="python" session="ex1"
+    import narwhals as nw
+    import pandas as pd
+
+    data = {'a': [1,2,3,4,5], 'b': [5,4,3,2,1], 'c': [10,20,30,40,50]}
+    pd_df = pd.DataFrame(data)
+
+    def approach_1(df_any):
+        df = nw.from_native(df_any)
+
+        # Pay attention to this next line using group_by()
+        df = df.group_by('a').agg((nw.col('b')+nw.col('c')).sum().alias('d'))
+        return nw.to_native(df)
+
+    print(approach_1(pd_df))
```

Comment:
Two tips in here which apply also for `approach_2`:

- We can use `@nw.narwhalify` and avoid `from|to_native` operation to display
- If we split the operation in multiple lines I think it is easier to see the difference between the two approaches

Putting it together

```suggestion
    @nw.narwhalify
    def approach_1(df):
        return (df
            .group_by('a')
            .agg(
                (nw.col('b')+nw.col('c')).sum().alias('d')
             )
        )
    print(approach_1(pd_df))
```